RamachanDraw/plot.py

```diff
@@ -1,43 +1,67 @@
-import os
-import matplotlib.pyplot as plt
+from os import sep as os_separator
+from os.path import realpath, exists
+from typing import Union
+
 import matplotlib.colors as mcolors
+import matplotlib.pyplot as plt
 import numpy as np
-from Bio.PDB import PDBParser, PPBuilder
 from pkg_resources import resource_stream
-from math import pi
+
+from .phi_psi import get_ignored_res
+from .fetch import get_file
 
 
-def plot(pdb_file, cmap='viridis', alpha=0.75, dpi=100, save=True, show=False, out='plot.png'):
-    batch_mode = [True if type(pdb_file) is list else False][0]
+def get_pdb_name(path: Union[str, list, tuple], remove_extension: bool = True, upper_case: bool = False) -> tuple:
+    """
+    Extracts the PDB file name(s)
+    :param path: path to extract name from
+    :param remove_extension: with(out) file extension
+    :param upper_case: convert output names to upper case
+    :return: Filenames as iterable
+    """
+    if isinstance(path, str):
+        path = [path]
 
-    def get_ignored_res(file: str):
-        x, y, ignored, output = [], [], [], {}
-        for model in PDBParser().get_structure(id=None, file=file):
-            for chain in model:
-                peptides = PPBuilder().build_peptides(chain)
-                for peptide in peptides:
-                    for aa, angles in zip(peptide, peptide.get_phi_psi_list()):
-                        residue = aa.resname + str(aa.id[1])
-                        output[residue] = angles
+    path_out = []
+    for p in path:
+        path_split = p.split(os_separator)
+        pdb_name = path_split[-1]
+        if remove_extension:
+            pdb_name_split = pdb_name.split(".")
+            if len(pdb_name_split) > 1:
+                pdb_name = '.'.join(pdb_name_split[:-1])
+        path_out.append(pdb_name.upper() if upper_case else pdb_name)
+    return tuple(path_out)
```

Comment:
This is a nice feature! :)

There may be a nicer way to do this, but for now it seems to get the job done.